src/coreclr/vm/threads.h

```diff
@@ -952,11 +952,6 @@ class Thread
     // in the object header to store it.
     DWORD                m_ThreadId;
 
-
-    // RWLock state
-    LockEntry           *m_pHead;
```

Comment:
Can we delete `LockEntry` completely, or at least more it to oletls.h that is its only use? 

We do not need to full definition of SOleTlsData. We only need the part up until the fields that we depend on. The rest can be deleted. (A separate question is whether it is a good idea to depend on SOleTlsData that is Windows internal impl detail.)

Reply:
It looks like the only portion of `SOleTlsData` that we use is the context pointer. Would it be possible for us to use [`CoGetObjectContext`[(https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-cogetobjectcontext) directly instead of going through the `SOleTlsData` object? Then we could remove it entirely.

Reply:
We would take some perf regression by replacing the shortcut with `CoGetObjectContext`. Also, we set the `SOleTlsData` field here: https://github.com/dotnet/runtime/blob/043ae8c50dbe1c7377cf5ad436c5ac1c226aef79/src/coreclr/vm/olecontexthelpers.cpp#L54 ... I am not sure what to do about that.

Reply:
Yeah I saw that. I wonder if we ever hit that path. My guess would be there was some bug in Windows that didn't set that sometimes so .NET Framework would do it. I really hope that path doesn't actually get hit in production...

At least since that's the only field we use, we could use [the official definition of the struct from the docs](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ms690269(v=vs.85)) for a slightly better experience.

Also, could we implement the fast path with our own thread-local variable instead of using this implementation detail?

Reply:
> At least since that's the only field we use, we could use [the official definition of the struct from the docs](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/legacy/ms690269(v=vs.85)) for a slightly better experience.

Yeah, I think it would make sense to do it in this PR. It should be a simple change.

I agree that it would be nice to clean it up further, but I am not sure whether it is worth spending time on it. It is guaranteed to break something obscure.